[Bug]: nil pointer dereference when setting "minimum_enabled_tls_protocol" to "TLS1_3"

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Provider Version

v1.15.3

### Terraform Version

v1.7.5

### Terraform Edition

Terraform Open Source (OSS)

### Current Behavior

Attempting to apply a plan, to an existing Atlas Cluster, which sets the minimum enabled TLS protocol to TLS 1.3, crashes the plugin. "TLS1_3" is an invalid value according to https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster#minimum_enabled_tls_protocol but should it cause a crash?

### Terraform configuration to reproduce the issue

```shell
resource "mongodbatlas_cluster" "cluster" {
  project_id             = var.project_id
  name                   = var.cluster_name
  mongo_db_major_version = var.mongodbversion
  cluster_type           = "REPLICASET"
  replication_specs {
        num_shards = 1
        regions_config {
        region_name     = var.region
        electable_nodes = 3
        priority        = 7
        read_only_nodes = 0
        }
  }
  cloud_backup                     = true
  auto_scaling_disk_gb_enabled = true
  provider_name                    = var.cloud_provider
  provider_instance_size_name  = "M10"
  advanced_configuration {
    minimum_enabled_tls_protocol         = "TLS1_3"
    javascript_enabled                   = false
  }

}
output "connection_strings" {
  value = mongodbatlas_cluster.cluster.connection_strings[0].standard_srv
}
```


### Steps To Reproduce

1. Run 'terraform plan' to plan to deploy TLS 1.3 minimum
2. Run 'terraform apply' to apply the plan
3. Expected: error message saying value is not valid
4. Actual: plugin crashes (output pasted below).

### Logs

```hcl
terraform plan -var-file=atlas_cluster.tfvars -out atlas.plan
mongodbatlas_cluster.cluster: Refreshing state... [id=Y2x1c3Rlcl9pZA==:NjYxM2NiNTAxNTdmNDU2ZTViYzJjM2M0-Y2x1c3Rlcl9uYW1l:Q2x1c3RlcjA=-cHJvamVjdF9pZA==:NjIwMTc1ODdjYWQ1NWYyNzJkNmYxYTE4-cHJvdmlkZXJfbmFtZQ==:QVdT]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # mongodbatlas_cluster.cluster will be updated in-place
  ~ resource "mongodbatlas_cluster" "cluster" {
        id                                      = "Y2x1c3Rlcl9pZA==:NjYxM2NiNTAxNTdmNDU2ZTViYzJjM2M0-Y2x1c3Rlcl9uYW1l:Q2x1c3RlcjA=-cHJvamVjdF9pZA==:NjIwMTc1ODdjYWQ1NWYyNzJkNmYxYTE4-cHJvdmlkZXJfbmFtZQ==:QVdT"
        name                                    = "Cluster0"
        # (32 unchanged attributes hidden)

      ~ advanced_configuration {
          ~ minimum_enabled_tls_protocol         = "TLS1_2" -> "TLS1_3"
            # (8 unchanged attributes hidden)
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Saved the plan to: atlas.plan

To perform exactly these actions, run the following command to apply:
    terraform apply "atlas.plan"
root@terraform:~/tf-atlas# terraform apply "atlas.plan"
mongodbatlas_cluster.cluster: Modifying... [id=Y2x1c3Rlcl9pZA==:NjYxM2NiNTAxNTdmNDU2ZTViYzJjM2M0-Y2x1c3Rlcl9uYW1l:Q2x1c3RlcjA=-cHJvamVjdF9pZA==:NjIwMTc1ODdjYWQ1NWYyNzJkNmYxYTE4-cHJvdmlkZXJfbmFtZQ==:QVdT]
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-mongodbatlas_v1.15.3 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x161baab]

goroutine 38 [running]:
github.com/mongodb/terraform-provider-mongodbatlas/internal/service/cluster.resourceMongoDBAtlasClusterUpdate({0x2216ff8, 0xc00047f800}, 0xc000110e80, {0x19026c0, 0xc00056e680})
        github.com/mongodb/terraform-provider-mongodbatlas/internal/service/cluster/resource_cluster.go:931 +0x15cb
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0x2216ff8?, {0x2216ff8?, 0xc00047f800?}, 0xd?, {0x19026c0?, 0xc00056e680?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:807 +0x7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000726fc0, {0x2216ff8, 0xc00047f800}, 0xc0009a7380, 0xc00093fb80, {0x19026c0, 0xc00056e680})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:919 +0x83a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000011b60, {0x2216ff8?, 0xc00047f650?}, 0xc000a5a820)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/grpc_provider.go:1078 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange({{0x2222800?, 0xc000011b60?}}, {0x2216ff8, 0xc00047f650}, 0x0?)
        github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5to6server/tf5to6server.go:47 +0x54
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0xc00065e1c0, {0x2216ff8?, 0xc00047ec90?}, 0xc000a5a7d0)
        github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf6muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc0006485a0, {0x2216ff8?, 0xc00047e2d0?}, 0xc000808000)
        github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/tf6server/server.go:846 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler({0x1b7e7e0, 0xc0006485a0}, {0x2216ff8, 0xc00047e2d0}, 0xc00093e000, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:518 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00011d800, {0x2216ff8, 0xc00047e240}, {0x221f160, 0xc00070ad00}, 0xc00093c000, 0xc00047f410, 0x30899b8, 0x0)
        google.golang.org/grpc@v1.62.1/server.go:1386 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc00011d800, {0x221f160, 0xc00070ad00}, 0xc00093c000)
        google.golang.org/grpc@v1.62.1/server.go:1797 +0xe87
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.62.1/server.go:1027 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 34
        google.golang.org/grpc@v1.62.1/server.go:1038 +0x125

Error: The terraform-provider-mongodbatlas_v1.15.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

Not sure if related, but have a nil pointer dereference too on the same version, but with the IdP resource:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x16a79cf]

goroutine 38 [running]:
github.com/mongodb/terraform-provider-mongodbatlas/internal/service/federatedsettingsidentityprovider.oldSDKUpdate(***0x22170a0, 0xc000201e30***, ***0xc000602fc0, 0x18***, ***0xc000602ff0, 0x14***, 0xc000653e00, ***0x19026c0, 0xc00064e3e0***)
	github.com/mongodb/terraform-provider-mongodbatlas/internal/service/federatedsettingsidentityprovider/resource_federated_settings_identity_provider.go:409 +0x34f
github.com/mongodb/terraform-provider-mongodbatlas/internal/service/federatedsettingsidentityprovider.resourceMongoDBAtlasFederatedSettingsIdentityProviderUpdate(***0x22170a0, 0xc000201e30***, 0xc000653e00, ***0x19026c0, 0xc00064e3e0***)
	github.com/mongodb/terraform-provider-mongodbatlas/internal/service/federatedsettingsidentityprovider/resource_federated_settings_identity_provider.go:291 +0x209
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc00060d500, ***0x2216ff8, 0xc000909350***, 0xc000653e00, ***0x19026c0, 0xc00064e3e0***)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.[33](https://github.com/zeals-co-ltd/infra-mongo-atlas/actions/runs/8626459940/job/23644694849#step:15:34).0/helper/schema/resource.go:812 +0x119
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00060d500, ***0x2216ff8, 0xc000909350***, 0xc0007cb6c0, 0xc000653c80, ***0x19026c0, 0xc00064e3e0***)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:919 +0x83a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000010240, ***0x2216ff8?, 0xc0009091d0?***, 0xc000641770)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/grpc_provider.go:1078 +0xd5c
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ApplyResourceChange(***0x2222800?, 0xc000010240?***, ***0x2216ff8, 0xc0009091d0***, 0x0?)
	github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5to6server/tf5to6server.go:47 +0x54
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ApplyResourceChange(0xc000710000, ***0x2216ff8?, 0xc000908f00?***, 0xc000641720)
	github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf6muxserver/mux_server_ApplyResourceChange.go:[36](https://github.com/zeals-co-ltd/infra-mongo-atlas/actions/runs/8626459940/job/23644694849#step:15:37) +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ApplyResourceChange(0xc0006fe5a0, ***0x2216ff8?, 0xc000908540?***, 0xc000200930)
	github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/tf6server/server.go:846 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ApplyResourceChange_Handler(***0x1b7e7e0, 0xc0006fe5a0***, ***0x2216ff8, 0xc000908540***, 0xc000653[38](https://github.com/zeals-co-ltd/infra-mongo-atlas/actions/runs/8626459940/job/23644694849#step:15:39)0, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.22.1/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:518 +0x1a6
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001b7800, ***0x2216ff8, 0xc0009084b0***, ***0x221f160, 0xc0007ba4e0***, 0xc0007df440, 0xc000847410, 0x30899b8, 0x0)
	google.golang.org/grpc@v1.62.1/server.go:1386 +0xdf8
google.golang.org/grpc.(*Server).handleStream(0xc0001b7800, ***0x221f160, 0xc0007ba4e0***, 0xc0007df4[40](https://github.com/zeals-co-ltd/infra-mongo-atlas/actions/runs/8626459940/job/23644694849#step:15:41))
	google.golang.org/grpc@v1.62.1/server.go:1797 +0xe87
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.62.1/server.go:1027 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 68
	google.golang.org/grpc@v1.62.1/server.go:1038 +0x125

Error: The terraform-provider-mongodbatlas_v1.15.3 plugin crashed!
```

--- a/.changelog/2139.txt
+++ b/.changelog/2139.txt
@@ -1,3 +1,3 @@
 ```release-note:bug
-resource/mongodbatlas_cluster: Fixes nil pointer dereference if `advanced_configuration` update fails in `mongodbatlas_cluster`
+resource/mongodbatlas_cluster: Fixes nil pointer dereference if `advanced_configuration` update fails
 ```

--- a/.changelog/2139.txt
+++ b/.changelog/2139.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/mongodbatlas_cluster: Fixes nil pointer dereference if `advanced_configuration` update fails in `mongodbatlas_cluster`
+```

--- a/internal/service/cluster/resource_cluster.go
+++ b/internal/service/cluster/resource_cluster.go
@@ -12,17 +12,19 @@ import (
 	"strings"
 	"time"
 
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/spf13/cast"
+
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/constant"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/validate"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/advancedcluster"
-	"github.com/spf13/cast"
-	matlas "go.mongodb.org/atlas/mongodbatlas"
 )
 
 const (
@@ -926,9 +928,9 @@ func resourceMongoDBAtlasClusterUpdate(ctx context.Context, d *schema.ResourceDa
 		if aclist, ok1 := ac.([]any); ok1 && len(aclist) > 0 {
 			advancedConfReq := expandProcessArgs(d, aclist[0].(map[string]any))
 			if !reflect.DeepEqual(advancedConfReq, matlas.ProcessArgs{}) {
-				argResp, _, err := conn.Clusters.UpdateProcessArgs(ctx, projectID, clusterName, advancedConfReq)
+				_, _, err := conn.Clusters.UpdateProcessArgs(ctx, projectID, clusterName, advancedConfReq)
 				if err != nil {
-					return diag.FromErr(fmt.Errorf(errorAdvancedConfUpdate, clusterName+argResp.DefaultReadConcern, err))
+					return diag.FromErr(fmt.Errorf(errorAdvancedConfUpdate, clusterName, err))
 				}
 			}
 		}

